### ISSUE TITLE
[dtensor] ops coverage tracker

### ISSUE THREAD

**Issue by wanchaol** [pytorch/pytorch#119930]
This issue is the overall tracker of the ops we are implementing currently and plan to implement soon, when we plan to add new ops we'll also try to update this issue to keep things in sync. 

### Current supported ops consolidation
replace the existing ops that implemented in rule with op strategy to increase op sharding coverage
- [ ] aten.slice_backward.default, 
- [ ] aten.bernoulli.default/aten.bernoulli_.float
- [ ] aten.split.Tensor, aten.split_with_sizes.default, aten.split_with_sizes_copy.default
- [ ] aten.cat.default
- [ ] aten.index.Tensor
- [ ] aten.index_select.default
- [ ] aten.squeeze.default, aten.squeeze.dim
- [ ] aten.view.default, aten._unsafe_view.default, aten.reshape.default
- [ ] aten.unsqueeze.default
- [ ] aten.expand.default
- [ ] aten.permute.default
- [ ] aten.transpose.int, aten.t.default
- [ ] aten.repeat.default
- [ ] aten.convolution.default
- [ ] aten.convolution_backward.default

### Ops to be implemented (planned)
- [ ] aten._scaled_dot_product_flash_attention.default
- [ ] aten._scaled_dot_product_flash_attention_backward.default
- [ ] complex ops

**Comment by Vandermode** [comment]
Hi @wanchaol , thanks for the great work!

I was wondering if that is possible to add user-customized partition rules to certain operators e.g., `torch.fft.fft `
Specifically, can we extend FFT to dtensor op without diving into the cpp implementation? 
Note such implementation is provided in jax (https://jax.readthedocs.io/en/latest/jax.experimental.custom_partitioning.html) to make it work in shared arrays.
 
Thank you for your patience.

**Comment by awgu** [comment]
https://github.com/pytorch/pytorch/issues/119930
https://github.com/pytorch/pytorch/issues/132016
https://github.com/pytorch/pytorch/issues/132017

**Comment by tianyu-l** [comment]
Hi @Vandermode , we've added the support for defining op partition/sharding rules in DTensor (in https://github.com/pytorch/pytorch/pull/131108). Please feel free to try out the API and let us know any feedbacks you may have.

**Comment by XilunWu** [comment]
https://github.com/pytorch/pytorch/pull/149106 is another tracker for DTensor op coverage for some important models.

**Comment by stmcgovern** [comment]
Hi @wanchaol , in working on this [PR](https://github.com/pytorch/pytorch/pull/164995) involving a sharding strategy, I discovered this issue. I'm very interested in contributing to this effort to support ops with a sharding strategy. Is this issue up to date? Does the team have any suggestions on good ops to try to contribute? Thanks!

cc @XilunWu @tianyu-l @zpcore
